.github/workflows/tests.yml

```
@@ -8,13 +11,59 @@ on:
   workflow_dispatch:
 
 jobs:
-  unittest-and-lint:
-    runs-on: ${{matrix.os}}
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+
+      - name: Setup up Linter
+        run: |
+          pip install toml yapf
+
+      - name: Lint code formatting
+        # make yapf fail if any file needs to be formatted
+        run: |
+          yapf -r --diff examples/ odxtools/ tests/
```

in order to be consistent with #119, you need to run `isort` after that: YAPF does not sort the imports according to PEP8 and it also changes the indentation of multi-line imports (I like the way isort does it better...)

```suggestion
          yapf -r --diff examples/ odxtools/ tests/
          isort -l 100 -m VERTICAL odxtools/ tests/ examples/
```

---

BTW: shouldn't we let the step fail if the source changes as a result of the reformatting action?

---

(this can be done using

```suggestion
          yapf -r -i odxtools/ tests/ examples/
          isort -l 100 -m VERTICAL odxtools/ tests/ examples/
          git diff --exit-code 2>&1 > /dev/null
```
)

---

The `--diff` option of `yapf`is simpler: It returns a non-zero return code in case something is reformatted and is supposed to be used in CIs according to the docs.

But you are right, isort is missing. I guess there is also a check option, otherwise I will go for the `git diff`